sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authroisation/SecurityContextComponent.java

```diff
@@ -0,0 +1,18 @@
+package uk.ac.ox.ndph.mts.sample_service.security.authroisation;
+
+import com.microsoft.azure.spring.autoconfigure.aad.UserPrincipal;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+/**
+ * Security context service to get data from the security context
+ */
+@Component
+public class SecurityContextComponent {
+
+    public UserPrincipal getUserPrincipal() {
```

Comment:
Does this really need to be an ...aad.UserPrincipal ? It doesn't look like you are using any of the specific methods. Also, you should always check the type returned by SecurityContextHolder before casting it.

Reply:
I have change this function to return the user id . And added a todo to change this implementation within the story to move to Spring Security libs: https://ndph-arts.atlassian.net/browse/ARTS-591